cas-server-support-pac4j/src/main/java/org/jasig/cas/support/pac4j/web/flow/ClientAction.java

```diff
@@ -189,9 +192,8 @@ protected void prepareForLoginPage(final RequestContext context) {
 
         // save parameters in web session
         final Service service = (Service) context.getFlowScope().get(SERVICE);
-        if (service != null) {
-            session.setAttribute(SERVICE, service);
-        }
+        logger.debug("save service: {}", service);
```

Comment:
This is a fix for a small bug I found when testing: when you play a first authentication for a service and then a second one without a service, it was not working as the ```null``` service was not saved (and thus restored).